src/win/util.c

```diff
@@ -484,7 +484,7 @@ uint64_t uv__hrtime(double scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  return (uint64_t) ((double) counter.QuadPart * hrtime_interval_ * scale);
+  return (uint64_t) (counter.QuadPart * (uint64_t) scale / hrtime_interval_);
```

Comment:
I doubt that the loss of precision from double arithmetic will ever matter in any real world scenarios, but let's not get stuck in this argument :) Certainly, using 128-bit integer math and doing multiplication before division will preserve all the precision of QPC values, so I suggest you implement it that way.

Another approach could be to implement both calculations starting from the same values returned by QPC, and measure the error of the floating point version over time. This would be just so we can satisfy our curiosity, and I don't think it is an absolute necessity once you have the 128-bit int implementation.

Thanks!